x/gamm/keeper/pool.go

```diff
@@ -284,3 +285,15 @@ func (k Keeper) setStableSwapScalingFactors(ctx sdk.Context, poolId uint64, scal
 
 	return k.setPool(ctx, stableswapPool)
 }
+
+// convertToCFMMPool converts PoolI to CFMMPoolI by casting the input.
+// Returns the pool of the CFMMPoolI or error if the given pool does not implement
+// CFMMPoolI.
+// nolint: unused
```

Comment:
Note: removing this is tracked in https://github.com/osmosis-labs/osmosis/issues/3556